improver/lightning.py

```diff
@@ -204,3 +205,215 @@ def latitude_to_threshold(
         rescale(latitude, (50.0, 10), (midlatitude, tropics), clip=True),
         rescale(latitude, (-50.0, -10), (midlatitude, tropics), clip=True),
     )
+
+
+class LightningMultivariateProbability_USAF2024(PostProcessingPlugin):
+    """
+    The algorithm outputs the probability of at least one lightning strike within
+    20 km of the location in a three hour period. The probabilities are calculated
+    for each individual ensemble member, and then they are averaged for the ensemble
+    probability forecast.
+
+    Inputs:
+    Convective Available Potential Energy (CAPE in J/kg)
+    Lifted Index (liftind in K)
+    Precipitable Water (pwat in kg m-2 or mm. This is used as mm in the regression equations)
+    Convective Inhibition (CIN in J/kg)
+    3-hour Accumulated Precipitation (apcp in kg m-2 or millimetres)
+    Outputs:
+    20 km lightning probability over the valid time of the accumulated precipitation
+
+
+    Description of the algorithm:
+
+    Regression equation when CAPE and APCP are greater than zero:
+    lprob= cape*APCP
+    lprob=0.13*alog(lprob+0.7)+0.05
+
+    If APCP is very low, a separate regression equation is used to predict lightning probability.
+    The definition of “very low” is raised slightly when PWAT values are high. This is because
+    the model often produces showery precipitation that doesn’t access the actual instability in
+    very moist environments:
+    lprob_noprecip=CAPE/(CIN+100.0)
+    lprob_noprecip=0.025*alog(lprob_noprecip+0.31)+0.03
+    APCP=APCP-(PWAT/1000)
+
+    IF APCP is less than 0.01 THEN lprob=lprob_noprecip
+    If there is no CAPE but the atmosphere is “close” to unstable, lightning does sometimes occur,
+    especially when heavy precipitation may have stabilized the atmosphere in the model. Unstable
+    values of lifted index are positive here:
+    LIFTIDX=LIFTIDX+4.0
+    IF LIFTIDX is less than 0 THEN LIFTIDX=0.0
+    IF CAPE is less than 0 THEN lprob =0.2*(LIFTIDX*APCP)^0.5
```

Comment:
I'm getting pretty picky, but below you define it as `<=`, so either here or there we should make these the same.

Reply:
Got it, defined it as <= here for CAPE to match the operations we actually perform in the code below.